numpy/_core/src/umath/string_ufuncs.cpp

```diff
@@ -665,6 +675,10 @@ init_ufunc(PyObject *umath, const char *name, const char *specname, int nin, int
 
     PyArray_DTypeMeta **dtypes = (PyArray_DTypeMeta **) PyMem_Malloc(
         (nin + nout) * sizeof(PyArray_DTypeMeta *));
+    if (dtypes == NULL) {
+        return -1;
```

Comment:
```suggestion
        PyErr_NoMemory();
        return -1;
```
(I think).  Allocation failures don't set an reror.